### PR TITLE
incorporate findsat_mrt into acstools

### PR DESCRIPTION
**Pull request by dvstark**
### Description

This pull request is incorporate newcode (findsat_mrt) into acstools. This code, described in ISR 2022-08, provides a new framework for identifying and masking satellite trails in ACS/WFC data. It should not impact any existing acstools programs.

### CODE REVIEW

**Comment by dvstark** [comment]
I'm aware of the odd line breaks with ^M at the end, and am trying to fix it

**Comment by dvstark** [comment]
end line character issue fixed. Code should now display normally

**Review by jryon — CHANGES_REQUESTED** [review]
Looking pretty good to me, but some changes are needed. I've left inline comments on specific items, but it's very likely I haven't covered everything.

One general concern I have is that it looks like you are making a lot of diagnostic plots, which I think is great, but the way it is written assumes users have interactive plotting enabled. Not sure we can/should assume that, so I'd suggest instead to save to disk every time a plotting function is called.

Looks like a typo in one of the doc filenames: `doc/source/utils_findsat_mrt.py` should be `doc/source/utils_findsat_mrt.rst`

Thanks for your hard work to integrate this into acstools!

File: acstools/findsat_mrt.py [hunk 1]
```diff
@@ -0,0 +1,1129 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+This module contains a class called trailfinder that is used to identify 
+satellite trails and/or other linear features in astronomical image data. To 
+accomplish this goal, the Median Radon Transform (MRT) is calculated for an image. 
+Point sources are then extracted from the MRT and filtered to yield a final 
+catalog of trails. These trails can then be used to create a mask.
+
+Example 1: Identificaation of trails in an ACS/WFC image, j97006j5q_flc.fits (4th extension)
```
[the remark this answers is not in the document]
Comment (dvstark) [reply acstools/findsat_mrt.py:10]: Regarding the interactive plotting, would it be ok to have python check if interactive mode is running, e.g.:
`import matplotlib as mpl
mpl.is_interactive()`
If that returns false, the plots can be written out to file, otherwise, they could appear as popups (I could also add an option to both plot on screen and save to disk)

File: acstools/findsat_mrt.py [hunk 2]
```diff
@@ -0,0 +1,1129 @@
...
+        self.save_catalog = save_catalog
+        self.save_diagnostic = save_diagnostic
+        self.save_mrt = save_mrt
+        self.save_mask = save_mask
+        
+        #plot image upon initialization
+        if (np.any(image) != None) & (self.plot == True):
+            self.plot_image()
+
+    def run_mrt(self,theta=None,threads=None):
```
[the remark this answers is not in the document]
Comment (dvstark) [reply acstools/findsat_mrt.py:214]: The idea here was flexibility and it being clear which attributes from trailfinder go into which functions. For instance, without the theta and threads keywords, it may not be obvious that these are things that can be adjusted. Is there a best practice here? If possible, I'd like to keep the keyword options, but if they end up being used, I could then update the relevant attributes of the satfinder class.

File: acstools/findsat_mrt.py [hunk 3]
```diff
@@ -0,0 +1,1129 @@
...
+        #run the routine to remove angles if any bad ranges are specified
+        if (self.source_list is not None) and (self.ignore_theta_range is not None):
+            self._remove_angles()
+            
+        #plot if set
+        if (self.plot==True) & (self.source_list is not None):
+            ax=self.plot_mrt()
+            for s in self.source_list:
+                ax.scatter(s['xcentroid'],s['ycentroid'],edgecolor='red',facecolor='none',s=100,lw=2)
+
```
[the remark this answers is not in the document]
Comment (dvstark) [reply acstools/findsat_mrt.py:456]: Added warning if no sources found.

File: acstools/findsat_mrt.py [hunk 4]
```diff
@@ -0,0 +1,1129 @@
...
+        ticks = np.arange(0,len(unique_vals)+1)
+        cax = plt.colorbar(mat, ticks=ticks)
+        #pdb.set_trace()
+        cax.ax.set_yticklabels(np.concatenate([unique_vals,[unique_vals[-1]+1]]))
+        cax.ax.set_ylabel('trail ID')
+        ax.set_title('Segmentation Mask')
+            
+            
+    def add_streak(self,endpoints,flux,width=3,psf_sigma=None):
+        '''
```
[the remark this answers is not in the document]
Comment (dvstark) [reply acstools/findsat_mrt.py:632]: Right, it is no longer needed so I removed it.

File: acstools/findsat_mrt.py [hunk 5]
```diff
@@ -0,0 +1,1129 @@
...
+
+        '''
+        
+        self.mask_bad_pixels(**kwargs)
+        self.subtract_background()
+        self.rebin(**kwargs)
+        
+
+def create_mrt_line_kernel(width,sigma,outfile=None,shape=(1024,2048),plot=False,theta=np.arange(0,180,0.5),threads=1):
+    '''
```
[the remark this answers is not in the document]
Comment (dvstark) [reply acstools/findsat_mrt.py:1010]: moved it as suggested

File: .gitattributes [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Set the default behavior, in case people don't have core.autocrlf set.
+* text=auto
+
+# Explicitly declare text files you want to always be normalized and converted
+# to native line endings on checkout.
+*.c text
+*.h text
+
+# Declare files that will always have CRLF line endings on checkout.
+*.sln text eol=crlf
```
Comment (pllim) [inline .gitattributes:10]: Is this Windows specific?

**Review by pllim — COMMENTED** [review]
tl;dr -- How does this relate to the existing satdet? Should that one be deprecated or something?

New doc pages not linked from top level index page?

There are no tests at all. How do we know code is working correctly and how to prevent future regression?

**Comment by pllim** [comment]
Also, CI failures need to be addressed.

File: acstools/findsat_mrt.py [hunk 7]
```diff
@@ -0,0 +1,1129 @@
...
+            plt.tight_layout()
+            plt.savefig('{}/{}_diagnostic.png'.format(output_dir,root))
+            if self.plot==False:
+                plt.close()
+
+        if save_catalog==True:
+            if self.source_list is not None:
+                self.source_list.write('{}/{}_catalog.fits'.format(output_dir,root),overwrite=True)
+            else:
+                #create an empty catalog and write that. It helps to have this for future analysis purposes even if it's empty
```
[the remark this answers is not in the document]
Comment (dvstark) [reply acstools/findsat_mrt.py:748]: I let an empty catalog be written because it made it clear that the image was inspected but nothing found, as opposed to an image that was never inspected. If I did not write a catalog for images where no trails were found, I could make a note in the log, but I feel like this adds another layer of complexity to analysis downstream when merging all the catalog files. With that explanation, is it ok to leave the code as is? Happy to discuss alternatives!

File: acstools/findsat_mrt.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jryon) [reply acstools/findsat_mrt.py:10]: This solution sounds good to me.

File: acstools/findsat_mrt.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jryon) [reply acstools/findsat_mrt.py:214]: I searched around a bit, but couldn't find a specific best practice for this kind of thing. I'd be fine with keeping the keywords and updating the attributes if they are used.

File: acstools/utils_findsat_mrt.py [hunk 8]
```diff
@@ -0,0 +1,1047 @@
...
+    
+    
+    counter=1
+    for t in tbls:
+        if counter==1:
+            src = t
+            counter += 1
+        else:
+            #check that added sources are not recreations of already detected ones
+            keep = (np.zeros(len(t))+1).astype(bool)
```
[the remark this answers is not in the document]
Comment (dvstark) [reply acstools/utils_findsat_mrt.py:81]: this should be fine. It checks each source and sets the value to keep=false if its too close to an existing source

File: acstools/tests/test_wfc_findsat_mrt.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
+'''
+Test satellite trail detection and masking using findsat_mrt
+'''
+
+from acstools.findsat_mrt import wfc_wrapper
+from acstools.tests.helpers import BaseACSTOOLS
+import acstools as a
+package_path = a.__path__[0]
```
Comment (pllim) [inline acstools/tests/test_wfc_findsat_mrt.py:8]: Not sure what this is doing.
```suggestion
```

File: acstools/tests/test_wfc_findsat_mrt.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+import os
+
+#update all this with my code
+
+class test_findsat_mrt(BaseACSTOOLS):
```
Comment (pllim) [inline acstools/tests/test_wfc_findsat_mrt.py:13]:
```suggestion
class TestFindsatMRT(BaseACSTOOLS):
```

File: acstools/tests/test_wfc_findsat_mrt.py [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        inputfile = rootname + '_flc.fits'  
+        truthmaskfile = rootname + '_flc_mask_ref.fits'
+        truthcatalogfile = rootname + '_flc_catalog_ref.fits'
+
+        # Prepare input file.
+        self.get_input_file(inputfile, skip_ref=True)
+
+        wfc_wrapper(inputfile, binsize=4, extension=4,
+                    output_root='jc8m32j5q',
+                    output_dir = package_path + '/tests/data/input/',
```
Comment (pllim) [inline acstools/tests/test_wfc_findsat_mrt.py:30]: Use tmp_path if possible: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

File: acstools/tests/test_wfc_findsat_mrt.py [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+from acstools.tests.helpers import BaseACSTOOLS
+import acstools as a
+package_path = a.__path__[0]
+import os
+
+#update all this with my code
+
+class test_findsat_mrt(BaseACSTOOLS):
+    detector = 'wfc'
+    package_directory = os.path.dirname(os.path.abspath(__file__))
```
Comment (pllim) [inline acstools/tests/test_wfc_findsat_mrt.py:15]:
```suggestion
```

File: acstools/tests/test_wfc_findsat_mrt.py (hunk 11 above)
Comment (dvstark) [reply acstools/tests/test_wfc_findsat_mrt.py:29]: Prefer to leave as is for the following reason: I'm trying to use the "compare_outputs" code which appears to only look in the /data/input and /data/truth/ directories for the files to test. So it is easier if my code writes the output files to the /data/input/ directory.

Comment (pllim) [reply acstools/tests/test_wfc_findsat_mrt.py:29]: Do you really need `compare_outputs` though. I haven't looked at your code in depth but that test class was written for HSTCAL that only can write to "working directory".

Anyways, if you are adamant on this, you have to make sure Git ignore those output files so you don't accidentally check them into version control.

Comment (dvstark) [reply acstools/tests/test_wfc_findsat_mrt.py]: I modified the test to not use compare_outputs and use the temporary directory for the output files.

File: .gitattributes (hunk 6 above)
Comment (dvstark) [reply .gitattributes:10]: Not sure. I was having issues with github not understand line breaks in my files and this was a solution that seemed to work. I'm not on Windows, but Mac. Better way to accomplish this?

Comment (pllim) [reply .gitattributes:10]: What is `*.sln`?

Comment (dvstark) [reply .gitattributes]: I think I copied/pasted a "solution" to my problem without fully understanding, and my .gitattributes files actually may not be necessary. I removed it and the files seem to not have any linebreak problems. Anymore. I think this is resolved.

**Comment by dvstark** [comment]
At this point I believe I've addressed all comments and it passed all the automated checks. Some final responses to points below:

> tl;dr -- How does this relate to the existing satdet? Should that one be deprecated or something?

It's an alternative method, with some key advantages (e.g., sensitivity), but it also performs poorly under certain situations (very crowded field). It's possible satdet works better in crowded fields (although this has not been thoroughly tested), and so we prefer to leave it in place. It's up to the use which they want to use. 

> New doc pages not linked from top level index page?

Should be fixed now.

> There are no tests at all. How do we know code is working correctly and how to prevent future regression?

I added a test.

**Review by pllim — COMMENTED** [review]
Formatted HTML looks a little weird: https://acstools--176.org.readthedocs.build/en/176/findsat_mrt.html

In general, docstring should follow https://numpydoc.readthedocs.io/en/latest/format.html style guide. Section headings should be consistent throughout the package (see other modules for what level should you start at). "Hidden" functions (those with leading underscore) should not be exposed in API doc unless there is a good reason.

Code snippet continuation should use "..." and not ">>>".

File: doc/source/utils_findsat_mrt.py [hunk 13]
```diff
@@ -0,0 +1,9 @@
+.. _utils_findsat_mrt:
+
+**************************
+MRT-based Satellite Trails Detection Utilities
+**************************
```
Comment (pllim) [inline doc/source/utils_findsat_mrt.py:5]: Section underline should match the number of characters.

File: setup.cfg [hunk 14]
```diff
@@ -79,6 +83,8 @@ filterwarnings =
   ignore:Deprecation Warning.*timeout.*parameter no longer needed:UserWarning
   # This is from skimage import
   ignore:.*np\.bool8.*is a deprecated alias:DeprecationWarning
+  #often appears during MRT calculation but is inconsequential
+  ignore:All-NaN slice encountered:RuntimeWarning
```
Comment (pllim) [inline setup.cfg:87]: I feel like if you don't care about such warning, it should be handled properly during the calculation in the code, not as blanket ignore for test suite that also affect other modules. Please have a look to see if `np.seterr` is what you really want and use it as context manager to the affected calculation only.

File: acstools/tests/test_wfc_findsat_mrt.py [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+logging.basicConfig()
+LOG = logging.getLogger()
+LOG.setLevel(logging.INFO)
+
+
+class TestFindsatMRT(BaseACSTOOLS):
+
+    detector = 'wfc'
+
+    def test_wfc_wrapper(self, tmpdir):
```
Comment (pllim) [inline acstools/tests/test_wfc_findsat_mrt.py:21]: If you want to future-proof, `tmpdir` is now considered "legacy" and `tmp_path` should be used instead. See https://docs.pytest.org/en/stable/how-to/tmp_path.html#tmpdir-and-tmpdir-factory

File: acstools/utils_findsat_mrt.py [hunk 16]
```diff
@@ -0,0 +1,1253 @@
...
+except ImportError:
+    plt = None
+    warnings.warn('matplotlib not found, plotting is disabled',
+                  AstropyUserWarning)
+
+__taskname__ = "utils_findsat_mrt"
+__author__ = "David V. Stark"
+__version__ = "1.0"
+__vdate__ = "06-Dec-2022"
+__all__ = ['_round_up_to_odd', 'merge_tables', 'good_indices',
```
Comment (pllim) [inline acstools/utils_findsat_mrt.py:34]: Why include the "hidden" functions?

File: setup.cfg [hunk 17]
```diff
@@ -35,6 +35,8 @@ install_requires =
     astropy>=3.1
     requests
     PyYAML
+    photutils
```
Comment (pllim) [inline setup.cfg:38]: If you put these in `install_requires`, they are not needed in `extras_require`.

You should think about whether these are really required at install time (`install_requires`) or they are merely optional dependencies that are only needed if one needs to use one small part of the package (`extras_require`).

Comment (pllim) [reply setup.cfg:38]: Optional dependencies should only be imported locally when needed and nothing should crash horribly outside of the things that actually use them when they are not present.

File: doc/source/index.rst [hunk 18]
```diff
@@ -40,6 +40,8 @@ To install the development version of this package::
    acszpt
    acsphotcte
    satdet
+   findsat_mrt
+   utils_findsat_mrt
```
Comment (pllim) [inline doc/source/index.rst:44]: Why can't MRT only have one page?

File: acstools/utils_findsat_mrt.py [hunk 19]
```diff
@@ -0,0 +1,1253 @@
...
+    x0 = image.shape[1]/2-0.5
+    xarr = np.arange(image.shape[1])-x0
+    yarr = np.arange(image.shape[0])-y0
+    x, y = np.meshgrid(xarr, yarr)
+
+    # add a simple streak across the image.
+    image = add_streak(image, width, 1, rho=0, theta=90, psf_sigma=sigma)
+
+    # plot the image
+    if (plot is True) & (plt is not None):
```
Comment (pllim) [inline acstools/utils_findsat_mrt.py:1173]: Why are two conditions even needed for this check? 
```suggestion
    if plt is not None:
```

File: acstools/utils_findsat_mrt.py [hunk 20]
```diff
@@ -0,0 +1,1253 @@
...
+        fig2, ax2 = plt.subplots()
+        ax2.imshow(rt, aspect='auto', origin='lower')
+        ax2.set_xlabel('angle pixel')
+        ax2.set_ylabel('offset pixel')
+
+    # find the center of the signal by summing along each direction and finding
+    # the max.
+    rt_rho = np.nansum(rt, axis=1)
+    rt_theta = np.nansum(rt, axis=0)
+    fig, [ax1, ax2] = plt.subplots(1, 2)
```
Comment (pllim) [inline acstools/utils_findsat_mrt.py:1195]: Won't the rest of this function crash horribly from here onwards if you don't have matplotlib?

File: doc/source/utils_findsat_mrt.py (hunk 13 above)
Comment (dvstark) [reply doc/source/utils_findsat_mrt.py]: This file was actually out of date and should have been removed from the repository. Now done.

File: setup.cfg (hunk 14 above)
Comment (dvstark) [reply setup.cfg]: I now use warnings.filterwarnings for that specific warning, only inside the trailfinder class. It's very common due to how missing data are presented in the module, but it is inconsequential.

File: acstools/tests/test_wfc_findsat_mrt.py (hunk 15 above)
Comment (dvstark) [reply acstools/tests/test_wfc_findsat_mrt.py]: Response: I have replaced tmpdir with tmp_path

File: acstools/utils_findsat_mrt.py (hunk 16 above)
Comment (dvstark) [reply acstools/utils_findsat_mrt.py]: Didn't know the standard here. They have been removed, although I have generally modified which are "hidden".

File: setup.cfg (hunk 17 above)
Comment (dvstark) [reply setup.cfg:38]: Given that these are used at only specific times in the pipeline, I moved them to extras_require.

File: doc/source/index.rst (hunk 18 above)
Comment (dvstark) [reply doc/source/index.rst]: It can. I modified the rst files. utils_findsat_mrt.rst is no longer needed and has been deleted

File: acstools/utils_findsat_mrt.py (hunk 19 above)
Comment (dvstark) [reply acstools/utils_findsat_mrt.py:1273]: Response: "plt" says whether matplotlib exists, while plot says whether plots should be generated. I don't think those should necessarily be  merged into a single flag.

File: acstools/utils_findsat_mrt.py (hunk 20 above)
Comment (dvstark) [reply acstools/utils_findsat_mrt.py]: Yes, thanks for catching that. Those commands should be inside an if statement

**Comment by dvstark** [comment]
I have examined the docstring formatting link, as well as the STScI style guides, made the following general changes:
(1) Ensured all routines have single line description at the top. Any additional documentation comes after separated by an additional blank line
(2) replace >>> with ... for line continuations
(3) Adjusted headings around "Examples" to match other acstools routines (although existing docs don't always seem consistent)
(4) Took greater care when defining "hidden" functions (with _ before them). I may have not been doing this properly, so there are only very few of these now.
(5) More in-line commenting throughout
(6) Updated some class and variable names to be consistent with STScI style guide
(7) Not really a style issue, but I also added a routine to update the DQ array in an input flc/flt file, identically to how satdet does it.

I have tried to make the code match best standards. I am getting a new failure in CI / tests and CI / dev_deps_tests, which may be related to the issue discussed here: https://github.com/Lightning-AI/lightning/issues/16756. I'd appreciate someone more familiar with github workflows to take a look.

File: setup.cfg (hunk 17 above)
Comment (pllim) [reply setup.cfg:38]: I still see them listed here. Did you forget to remove?

**Comment by pllim** [comment]
Re: failing CI -- I ignored the warning in `master` branch, so if you rebase, the CI should pass unless your changes fail it.

File: setup.cfg (hunk 17 above)
Comment (dvstark) [reply setup.cfg]: Yes, I just forgot to commit. Fixed

**Comment by dvstark** [comment]
I rebased with master and modified the code to explicitly check for optional dependencies. All CI checks have passed

**Comment by pllim** [comment]
I just want to let you know that I will be doing a re-review soon. If you have a hard deadline to merge and/or release, please let me know. Thanks!

**Comment by dvstark** [comment]
No hard deadline, but certainly will be nice to wrap this up. Thanks in advance!

File: acstools/findsat_mrt.py [hunk 21]
```diff
@@ -0,0 +1,1362 @@
...
+    theta : ndarray, optional
+        Angles at which to calculate the MRT. The default is None, which
+        sets to np.arange(0,180,0.5).
+    kernels : list, optional
+        Paths to each kernel to be used for source finding in the MRT.
+        The default is None, which reverts to using the 3, 7, and 15 pixel
+        wide kernels included with this package.
+    mask_include_status: list, optional
+        List indicating trails with which status should be considered
+        when making the mask. The default is [2].
```
Comment (pllim) [inline acstools/findsat_mrt.py:190]: What does "2" mean? Is this explained somewhere?

**Comment by dvstark** [comment]
It’s referring to a status flag (ranging from 0-2) for each trail identified. They are explained in the docstring of filter_sources in utils_findsat_mrt.py, which is definitely too hard to find. I can add an explanation in the TrailFinder class.

--
David V. Stark
Staff Scientist II, Science Support
HST/ACS Team
Space Telescope Science Institute

From: "P. L. Lim" ***@***.***>
Reply-To: spacetelescope/acstools ***@***.***>
Date: Friday, March 24, 2023 at 4:16 PM
To: spacetelescope/acstools ***@***.***>
Cc: David Stark ***@***.***>, Author ***@***.***>
Subject: Re: [spacetelescope/acstools] incorporate findsat_mrt into acstools (PR #176)


@pllim commented on this pull request.

________________________________

In acstools/findsat_mrt.py<https://urldefense.com/v3/__https:/github.com/spacetelescope/acstools/pull/176*discussion_r1148015564__;Iw!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!zSoCVHnOonM6pE1naLlteg-l0sgcytk-RPdKLFiGVISH8JSrKUhubq-bX4ZI2UtjOneMfjb3PbjEulR6ZWaH9W0$>:

> +        Maximum streak width to allow. The default is 75 pixels.

+    buffer : int, optional

+        Size of cutout region extending perpendicular outward from a

+        streak. The default is 250 pixels on each side.

+    threshold : float, optional

+        Minimum S/N when extracting sources from the MRT. The default is 5.

+    theta : ndarray, optional

+        Angles at which to calculate the MRT. The default is None, which

+        sets to np.arange(0,180,0.5).

+    kernels : list, optional

+        Paths to each kernel to be used for source finding in the MRT.

+        The default is None, which reverts to using the 3, 7, and 15 pixel

+        wide kernels included with this package.

+    mask_include_status: list, optional

+        List indicating trails with which status should be considered

+        when making the mask. The default is [2].

What does "2" mean? Is this explained somewhere?

**Review by pllim — CHANGES_REQUESTED** [review]
I have a bunch of proposed changes over at https://github.com/dvstark/acstools/pull/1 . These include API changes and bug fixes as we discussed on Slack. I also applied changes from my earlier rounds of reviews. Reading the diff might drive you crazy, so it might be better for you to view the before/after side by side.

Since these include real bug fixes, if you run the unit test that you wrote with my changes, the result now disagrees with the "truth" you provided here. I cannot tell if that is okay or not, so please look at it carefully.

We can discuss via meeting the week after next if you are unable to review it on your own. Thank you for your continued patience!

**Comment by pllim** [comment]
A minor PEP 8 thingy to fix.

And also the link appears broken: https://www.clear.rice.edu/elec431/projects96/DSP/bpanalysis.html

**Comment by dvstark** [comment]
I'm suspecting that broken link was temporary; it seems to be working now. I'll fix the pep8 stuff and push again

File: acstools/findsat_mrt.py [hunk 22]
```diff
@@ -0,0 +1,1409 @@
...
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.kernels`.
+        The default is `None`, which reverts to using the 3-, 7-, and 15-pixel
+        wide kernels included with this package.
+    mask_include_status : list, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.mask_include_status`.
+        The default is ``[2]``.
+    plot : bool, optional
+        Plot all intermediate steps. When set, plots of the input image, MRT
+        with identified sources, and resulting image maks will be generated
+        after running :func:`acstools.findat_mrt.TrailFinder.__init__`:,
```
Comment (pllim) [inline acstools/findsat_mrt.py:177]: `__init__` is not exposed in API doc.
```suggestion
        after running the constructor,
```

File: acstools/findsat_mrt.py [hunk 23]
```diff
@@ -0,0 +1,1409 @@
...
+        :func:`acstools.findsat_mrt.TrailFinder.run_mrt`:,
```
Comment (pllim) [inline acstools/findsat_mrt.py:178]: Too many colons.
```suggestion
        :func:`acstools.findsat_mrt.TrailFinder.run_mrt`,
```

File: acstools/findsat_mrt.py [hunk 24]
```diff
@@ -0,0 +1,1409 @@
...
+        :func:`acstools.findsat_mrt.TrailFinder.find_mrt_sources`:,
+        :func:`acstools.findsat_mrt.TrailFinder.filter_sources`:,
+        and :func:`acstools.findsat_mrt.TrailFinder.make_mask`:. Users may also
+        generate these plots manually by calling
+        :func:`acstools.findsat_mrt.TrailFinder.plot_image`:,
+        :func:`acstools.findsat_mrt.TrailFinder.plot_mrt`:,
+        :func:`acstools.findsat_mrt.TrailFinder.plot_mask`:, and
+        :func:`acstools.findsat_mrt.TrailFinder.plot_segment`:. The default is
```
Comment (pllim) [inline acstools/findsat_mrt.py:186]:
```suggestion
        :func:`acstools.findsat_mrt.TrailFinder.find_mrt_sources`,
        :func:`acstools.findsat_mrt.TrailFinder.filter_sources`,
        and :func:`acstools.findsat_mrt.TrailFinder.make_mask`. Users may also
        generate these plots manually by calling
        :func:`acstools.findsat_mrt.TrailFinder.plot_image`,
        :func:`acstools.findsat_mrt.TrailFinder.plot_mrt`,
        :func:`acstools.findsat_mrt.TrailFinder.plot_mask`, and
        :func:`acstools.findsat_mrt.TrailFinder.plot_segment`. The default is
```

File: acstools/findsat_mrt.py [hunk 25]
```diff
@@ -0,0 +1,1409 @@
...
+    kernels : list, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.kernels`.
+        The default is `None`, which reverts to using the 3-, 7-, and 15-pixel
+        wide kernels included with this package.
+    mask_include_status : list, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.mask_include_status`.
+        The default is ``[2]``.
+    plot : bool, optional
+        Plot all intermediate steps. When set, plots of the input image, MRT
+        with identified sources, and resulting image maks will be generated
```
Comment (pllim) [inline acstools/findsat_mrt.py:176]:
```suggestion
        with identified sources, and resulting image mask will be generated
```

File: acstools/tests/test_wfc_findsat_mrt.py [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+        os.path.join('data', 'truth', f'{rootname}_flc_mask_ref.fits'),
+        package='acstools.tests', show_progress=False, remote_timeout=timeout)
+    actual = tmp_path / f'{rootname}_flc_mask.fits'
+
+    # If the machine does not have 8 cores, it will only use up to what is
+    # available. We save the catalog to make sure that code runs but we do not
+    # compare catalog.
+    WfcWrapper(inputfile, binsize=4, extension=4,
+               output_root=f'{rootname}_flc',
+               output_dir=tmp_path, max_width=75/2.,
```
Comment (pllim) [inline acstools/tests/test_wfc_findsat_mrt.py:27]:
```suggestion
               output_dir=tmp_path, max_width=37.5,
```

**Review by pllim — COMMENTED** [review]
Just some minor comments. I'll approve when they are addressed.

By the way, CI was green after I reran the failed jobs, so the linkcheck and socket failures were indeed transient.

**Comment by dvstark** [comment]
All your suggested changes made sense so I committed them all (and sorry if this led to various email notifications from github after every commit)

**Review by pllim — APPROVED** [review]
Thanks! @jryon , do you want to re-review since you requested changes?

If you happen to merge first, please use the `Squash and Merge` button.

**Review by jryon — APPROVED** [review]
I'm happy with it now that @pllim and @dvstark are!

**Comment by pllim** [comment]
Merged. Congrats and thanks for your patience! 

@jryon , you are doing releases now, right?

**Comment by jryon** [comment]
Yep, added to my to-do list.

**Comment by dvstark** [comment]
Thanks both!
